src/components/connect.js

```diff
@@ -88,7 +90,7 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
           this.finalMapStateToProps(state, props) :
           this.finalMapStateToProps(state)
 
-        return checkStateShape(stateProps)
+        return checkStateShape(this.constructor.displayName, stateProps)
```

Comment:
Rather than pass `this.constructor.displayName` in many places, let’s just calculate `var connectDisplayName = ...` right away and close over it everywhere we used `this.constructor.displayName` before this PR. Then we would be able to close over it in `checkStateShape` as well without even passing it.